src/test/groovy/com/example/EndpointSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package com.example
+
+import org.springframework.http.HttpStatus
+import spock.lang.Unroll
+
+class EndpointSpec extends IntegrationSpec {
+
+    def "should return list of cards with three elements"() {
+        when:
+        def response = restTemplate.getForEntity(localUrl("/get-cards"), Card[])
+
+        then:
+        response.statusCode == HttpStatus.OK
+        response.body.size() == 3
+    }
+
+    def "should return card with name 'Szkoła'"() {
+        when:
+        def response = restTemplate.getForEntity(localUrl("/find-card?word=Szkoła"), Card)
+
+        then:
+        response.statusCode == HttpStatus.OK
+        with(response.body) {
+            word == "Szkoła"
+            forbiddenWord == ["Uczelnia",
+                              "Nauczyciel",
+                              "Gimnazjum",
+                              "Podstawówka",
+                              "Uczeń"]
+        }
+    }
+
+    def "should throw exception when card with name 'Foo' does not exists"() {
+        when:
+        def response = restTemplate.getForEntity(localUrl("/find-card?word=Foo"), Card)
+
+        then:
+        response.statusCode == HttpStatus.INTERNAL_SERVER_ERROR
+    }
+
+    @Unroll
+    def "should return card with #name"(name, statusCode, cardWord, cardForbiddenWord) {
```

Comment:
moim zdaniem argumenty są spoko, bo testy powinny spełniać takie same standardy jak kod i jeśli byłoby ich siedem to jakaś statyczna analiza by pokazała naruszenie zasad. Czy są analizatory do tabelki ze spockowego `where` to niestety nie wiem ;)